### PR TITLE
PR for CA JS2

### PR DESCRIPTION
**Pull request by sulenchy**

### CODE REVIEW

**Comment by sulenchy** [comment]
@Miksel90 As far as the UI and the functionality are concerned, you have the following intact

- Login 
- register
- fetch all posts
- fetch a single post
- Search

And you have the following yet to be implemented:

- filter
- delete post
- update post

However, I noticed a bug where an auth user can access the login page. This should not be the case. Only unauthenticated user should access register and login page. And solution is to write a function that check if the user is auth or not and act accordingly

File: js/backToTopButton.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+window.onscroll = () => {
```
Comment (sulenchy) [inline js/backToTopButton.js]: This looks good to me

File: js/deletePost.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { API_BASE_URL } from "./const.mjs";
+
+const modal = document.getElementById("myModal");
+const deleteButton = document.getElementById("deletePostButton");
+
+async function deletePost(postId) {
```
Comment (sulenchy) [inline js/deletePost.js:10]: You have delete function? Ok. I can't seems to see that in the UI

File: js/deletePost.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  // console.log(url);
+
+  try {
+    const response = await fetch(url, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+      body: JSON.stringify({
```
Comment (sulenchy) [inline js/deletePost.js]: According to the API doc., this is incorrect. The id is meant to be passed to the endpoint in the url lik `/social/posts/<id>`

**Review by sulenchy — COMMENTED** [review]
Good job! so far. Please, try and complete the rest of the functionality ASAP.

File: js/deletePost.js (hunk 2 above)
Comment (Miksel90) [reply js/deletePost.js:10]: if you click the edit button on the profilepage posts its open a modal where you can edit and delete the post

**Pull request closed without merging**